v0/destinations/fbpixel/transform.js

```diff
@@ -0,0 +1,431 @@
+/* eslint-disable no-param-reassign */
+/* eslint-disable camelcase */
+const sha256 = require("sha256");
+const { CONFIG_CATEGORIES, MAPPING_CONFIG } = require("./config");
+const { EventType } = require("../../../constants");
+
+const {
+  constructPayload,
+  defaultPostRequestConfig,
+  defaultRequestConfig,
+  flattenJson
+} = require("../../util");
+
+function formatRevenue(revenue) {
+  return Number((revenue || 0).toFixed(2));
+}
+
+function getContentType(message, defaultValue, categoryToContent) {
+  const { options } = message;
+  if (options && options.contentType) {
+    return [options.contentType];
+  }
+
+  let { category } = message.properties;
+  if (!category) {
+    const { products } = message.properties;
+    if (products && products.length) {
+      category = products[0].category;
+    }
+  }
+  if (category) {
+    if (categoryToContent === undefined) {
+      categoryToContent = [];
+    }
+    const mapped = categoryToContent;
+    const mappedTo = mapped.reduce((filtered, map) => {
+      if (map.from === category) {
+        filtered = map.to;
+      }
+      return filtered;
+    }, "");
+    if (mappedTo.length) {
+      return mappedTo;
+    }
+  }
+  return defaultValue;
+}
+
+function handleOrder(message, categoryToContent) {
+  const { products } = message.properties;
+  const value = formatRevenue(message.properties.revenue);
+  const contentType = getContentType(message, "product", categoryToContent);
+  const contentIds = [];
+  const contents = [];
+  const { category } = message.properties;
+
+  for (let i = 0; i < products.length; i += 1) {
+    const pId = products[i].product_id || products[i].sku || products[i].id;
```

Comment:
put a utility function in EcommMappings.js to  getProductId(product)

Reply:
we are not handling "" for pid as in other cases.
let's have something uniform ?

Reply:
done